lib/auth/googleauth.ts

```diff
@@ -0,0 +1,592 @@
+/**
+ * Copyright 2013 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { exec } from 'child_process';
+import Compute from './computeclient';
+import IAMAuth from './iam';
+import JWTAccess from './jwtaccess';
+import JWTClient from './jwtclient';
+import JWT from './jwtclient';
+import OAuth2 from './oauth2client';
+import UserRefreshClient from './refreshclient';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as util from 'util';
+import DefaultTransporter from '../transporters';
+
+export default class GoogleAuth {
+
+  public transporter: DefaultTransporter;
+
+  /**
+   * Caches a value indicating whether the auth layer is running on Google Compute Engine.
+   * @private
+   */
+  public _isGCE = false;
```

Comment:
I know :(  It had to be public because unit tests checked the value.  I went ahead and made them private, then added public getters.  In reality we need this:
https://github.com/Microsoft/TypeScript/issues/5228